### ISSUE TITLE
Application lock pops up multiple times over locked app

### ISSUE THREAD

**Issue by nikolamilosevic86** [nikolamilosevic86/owasp-seraphimdroid#36]
When I have blocked an app, and I click on its icon, SeraphimDroid pops up to enter the code, but when I do, it does not work. When I went and clicked the SeraphimDroid icon , I entered the code, then the App Lock module appeared and from there I was able to unblock the app I wanted. 

Bug seems not always reproducible, but it needs deeper look how and why it appears and fixing.

**Comment by furquan1993** [comment]
I faced this issue while I was developing the app lock for seraphimdroid.
This seems to be the case when you try to open a locked application and
don't enter the correct password but just leave the application with out
using the exit button. This might cause seraphimdroid to conjure the lock
screen more than once.

I don't have a setup right now, but I am sure someone can look into it and
have a fix for this behavior.

I will be happy to help in case someone needs it.

**Comment by bohrium272** [comment]
If I am understanding correctly, @nikolamilosevic86 , @furquan1993 , is this what happens:
click on a locked application -> (Seraphimdroid pops up with the lock screen) -> Do not enter the password correctly or do not enter at all -> Leave by either pressing back or home, not the exit on the screen -> Lock Screen comes up again. I hope I traced it correctly.

**Comment by nikolamilosevic86** [comment]
Yes, if lock screen comes up again multiple times that is the problem. I have seen cases when you enter correct pin and lock screen comes again and again for many times. It might be same issue, however, if you press home or back, app should close and pin screen should be gone as well.
